playbooks/common/openshift-master/config.yml

```diff
@@ -337,12 +361,54 @@
     with_items: openshift_master_named_certificates
     when: named_certs_specified | bool
 
-- name: Configure master instances
-  hosts: oo_masters_to_config
+- name: Configure first master instance
```

Comment:
> then calls the openshift_docker role (which is likely due for a rename soon)

there will still be a need for the openshift_docker role, but splitting out the OpenShift version handling, etc into a separate role would make a lot of sense.

> I would propose we make the goal a consistent "protect currently installed unless upgrading", rather than "keep cluster in sync", and make the latter the job of a specific playbook for updating the cluster to latest version in the current stream. Seem reasonable?

I think my hangup here is what determines we are upgrading? This becomes even murkier when you consider the idealized stand-alone Ansible role use case:
- The input to the role specifies a specific version (or release):
  - If that version or release is already installed:
    - If the version matches, then do nothing
    - If the version is an upgrade, then initiate upgrade for role
    - If the version is a downgrade, error
- No version specified to role:
  - If already installed, do nothing
  - If not already installed, install latest

In the situations we've discussed so far, we want to lock the version installed across the cluster in some way to avoid drift. If we do this by keying off the first master, then we can ensure that deployments managed by the stand-alone playbooks/roles would eventually converge on a single version. For this case, and ideal scenario would be that simply changing the version or release for their deployment would trigger an upgrade that would start with the first master and roll through the rest of the cluster.

Obviously, we are quite far from that being reality, but having to use some externally controlled toggle to control the behavior outside of that seems a bit off.

What if there was some comparison done between the user provided variable and the one derived from the master? If the host is getting release 3.2 from the user provided variable and the master is telling us that we should be installing 3.3, then maybe some other behavior is triggered? Possibly failure or do skip package changes?